apisix/core/etcd.lua

```diff
@@ -15,11 +15,38 @@
 -- limitations under the License.
 --
 local fetch_local_conf = require("apisix.core.config_local").local_conf
-local etcd = require("resty.etcd")
-local clone_tab = require("table.clone")
+local etcd             = require("resty.etcd")
+local clone_tab        = require("table.clone")
+local io               = io
+local type             = type
+local ipairs           = ipairs
+local string           = string
+local tonumber         = tonumber
 
 local _M = {version = 0.1}
 
+local prefix_v3 = {
+    ["3.5"] = "/v3",
+    ["3.4"] = "/v3",
+    ["3.3"] = "/v3beta",
+    ["3.2"] = "/v3alpha",
+}
+
+
+-- TODO: Default lua-resty-etcd version auto-detection is broken, so directly get version from cmd
+--          we don't need to call this so many times, need to save it in some place
+local function etcd_version_from_cmd()
+    local cmd = "export ETCDCTL_API=3 && etcdctl version"
```

Comment:
Is user must install `etcdctl` in the APISIX node?

Reply:
etcdctl would be installed alongside with etcd

Reply:
If we only fetch the etcd version, we can use `etcd` v2 library, it should work fine.

Reply:
here is an example code: https://github.com/api7/lua-resty-etcd/blob/master/lib/resty/etcd.lua#L17